src/backend/distributed/commands/alter_table.c

```diff
@@ -1589,7 +1625,36 @@ ReplaceTable(Oid sourceId, Oid targetId, List *justBeforeDropCommands,
 		ExecuteQueryViaSPI(query->data, SPI_OK_INSERT);
 	}
 
-	List *ownedSequences = getOwnedSequences(sourceId);
+	/* We need to drop sequence dependencies for identities to preserve their
+	 * last value on the workers if it's a undistribute table conversion
+	 */
+	if (conversionType == UNDISTRIBUTE_TABLE)
```

Comment:
This is not just an `undistribute_table` problem:
``` SQL
-- from coordinator
CREATE TABLE color ( color_id BIGINT GENERATED ALWAYS AS IDENTITY UNIQUE, color_name VARCHAR NOT NULL );
select create_distributed_table('color', 'color_id');

-- from worker
INSERT INTO color(color_name) VALUES ('Red');

-- from coordinator
select alter_distributed_table('color', shard_count := 6);

-- from worker
INSERT INTO color(color_name) VALUES ('Red');
ERROR:  duplicate key value violates unique constraint "color_color_id_key_102180"
DETAIL:  Key (color_id)=(281474976710657) already exists.
```

Reply:
probably applies to all forms of ReplaceTable